lib/requestContextPlugin.js

```diff
@@ -10,7 +10,7 @@ function plugin(fastify, opts, next) {
   fastify.decorate('requestContext', requestContext)
   fastify.decorateRequest('requestContext', requestContext)
 
-  fastify.addHook('onRequest', (req, res, done) => {
+  fastify.addHook(opts.hook || 'preValidation', (req, res, done) => {
```

Comment:
Would it be a semver major this change?

***continue
if the opts.hook is an array, it won't works, so we need to iterate over the array.
Why we should add the same hook on multiple lifecycle events?

Reply:
Yes, will have to do major version bump.

Reply:
Hopefully there is a way to fix losing context from onRequest stage, though, but worst case scenario, initting ASL only on preValidation upwards can be made canonic.